### ISSUE TITLE
Return the Job Identifier on dispatch

### ISSUE THREAD

**Issue by egorovpavel** [cavai-research/Adonis-Queue#35]
It would be nice to get a some kind of reference to a job after it was dispatched, so we can track its status.

so instead of "fire and forget"

```ts
await MyJob.dispatch({config})
```

we could do something with the job

```ts
const job = await MyJob.dispatch({config})
```

---
Thanks for the proper "redisless" queue!

**Comment by McSneaky** [comment]
True! That would be really nice indeed
Did initial test real fast. Need to return from two places
- https://github.com/cavai-research/Adonis-Queue/blob/main/src/Dispatcher.ts#L55
- https://github.com/cavai-research/Adonis-Queue/blob/main/src/Drivers/Database.ts#L17

Also need to update TS typings and would be cool to have some unified return across drivers for job creation
I wonder, if just job ID would be enuf to return or should return more info? 🤔 
Like
```ts
const job = await MyJob.dispatch({ foo: 'bar' })
console.log(job)
// 10

// vs
console.log(job)
{
  id: 10,
  class_path: '/foo/bar/MyJob.ts',
  payload: { foo: 'bar' }
}
```
For just tracking, ID itself would be enuf, but would there be cases for other info to be returned? 🤔

**Comment by McSneaky** [comment]
Just realized, can start with returning:
```ts
{
  id: 10
}
```
And then can add extra return data when needed in future without breaking changes

**Comment by McSneaky** [comment]
Released as version 3.0.0. Breaking change for people who wrote custom drivers. They'll need to return job ID too now
